From dd298a4bf92465505b6feedecd4c3de41ead01ba Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <yesshedi@gmail.com>
Date: Sat, 9 May 2026 17:31:17 +0530
Subject: [PATCH 14/18] Fix -Wdiscarded-qualifiers error with glibc-2.43

util/bpf-loader.c: In function 'config_bpf_program':
util/bpf-loader.c:588:27: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]                                                                                                            588 |                 char *s = strchr(main_str, ':');

Signed-off-by: Shreenidhi Shedi <yesshedi@gmail.com>
---
 tools/perf/util/bpf-loader.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/perf/util/bpf-loader.c b/tools/perf/util/bpf-loader.c
index f4adeccdbbcb..7f4e774a3435 100644
--- a/tools/perf/util/bpf-loader.c
+++ b/tools/perf/util/bpf-loader.c
@@ -585,7 +585,7 @@ config_bpf_program(struct bpf_program *prog)
 		goto errout;
 
 	if (is_tp) {
-		char *s = strchr(main_str, ':');
+		const char *s = strchr(main_str, ':');
 
 		priv->is_tp = true;
 		priv->sys_name = strndup(main_str, s - main_str);
-- 
2.54.0

